### PR TITLE
98 support router tasks

### PR DESCRIPTION
**Pull request by jackschofield23**
Changes required for #98

### Description

Adds support for router tasks

### Status
**DONE**

### Types of changes
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] All tests passed locally.
- [ ] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.
- [ ] User guide updated.
- [ ] I have updated the changelog
- [x] Any new files have copyright headers
- [x] Code coverage above 70%
- [x] TODO items resolved

### CODE REVIEW

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs [hunk 1]
```diff
@@ -361,6 +345,20 @@ private async Task<bool> HandleTaskDestinations(WorkflowInstance workflowInstanc
 
             foreach (var taskExec in taskExecutions)
             {
+                if (string.Equals(taskExec.TaskType, TaskTypeConstants.RouterTask, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    processed &= await HandleTaskDestinations(workflowInstance, workflow, taskExec, correlationId);
+
+                    if (!processed)
```
Comment (neildsouth) [inline src/WorkflowExecuter/Services/WorkflowExecuterService.cs:352]: small point, but I prefer `if (processed is false)`

File: src/WorkflowManager/Program.cs [hunk 2]
```diff
@@ -120,7 +120,7 @@ private static void Main(string[] args)
 
             if (args.Length == 0)
             {
-                host.Services.GetService<TaskManager.TaskManager>().StartAsync(System.Threading.CancellationToken.None).RunSynchronously();
+                //host.Services.GetService<TaskManager.TaskManager>().StartAsync(System.Threading.CancellationToken.None).RunSynchronously();
```
Comment (lillie-dae) [inline src/WorkflowManager/Program.cs:123]: is this supposed to be commented?

Comment (jackschofield23) [reply src/WorkflowManager/Program.cs:123]: good spot, fixed

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs [hunk 3]
```diff
@@ -374,6 +372,22 @@ private async Task<bool> HandleTaskDestinations(WorkflowInstance workflowInstanc
             return processed;
         }
 
+        private async Task<bool> HandleTaskDestinations(WorkflowInstance workflowInstance, WorkflowRevision workflow, TaskExecution task, string correlationId)
+        {
+            var newTaskExecutions = await CreateTaskDestinations(workflowInstance, workflow, task.TaskId);
+
+            if (!newTaskExecutions.Any())
```
[the remark this answers is not in the document]
Comment (jackschofield23) [reply src/WorkflowExecuter/Services/WorkflowExecuterService.cs]: Changed.

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs (hunk 1 above)
Comment (jackschofield23) [reply src/WorkflowExecuter/Services/WorkflowExecuterService.cs]: Changed

**Review by neildsouth — APPROVED** [review]

**Pull request merged**